merge.go

```diff
@@ -79,6 +88,22 @@ func (r *Repository) AnnotatedCommitFromRef(ref *Reference) (*AnnotatedCommit, e
 	return newAnnotatedCommitFromC(ptr, r), nil
 }
 
+func (r *Repository) AnnotatedCommitFromRevspec(spec string) (*AnnotatedCommit, error) {
+	crevspec := C.CString(spec)
+	defer C.free(unsafe.Pointer(crevspec))
+
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	var ptr *C.git_annotated_commit
+	ret := C.git_annotated_commit_from_revspec(&ptr, r.ptr, crevspec)
+	runtime.KeepAlive(r)
+	if ret < 0 {
+		return nil, MakeGitError(ret)
+	}
+	return newAnnotatedCommitFromC(ptr, r), nil
```

Comment:
The `runtime.KeepAlive()` call needs to come after our last reference to what we need to keep alive. The annotated commit goes into `newAnnotatedCommitFromC` and we need to make sure `r` stays alive until we return from this function.

Reply:
Fixed this and all other methods that have the same issue.